python/MOCK_AIS.py

```diff
@@ -140,6 +140,18 @@ def remove_boat_callback(self, msg):
     def gps_callback(self, msg):
         self.sailbot_lat = msg.lat
         self.sailbot_lon = msg.lon
+    
+    def check_new_num_ship(self, previous_num_ships):
+        updated_num_ships = rospy.get_param('num_ais_ships', default=5)
```

Comment:
len(self.ships) is number of ships currently
updated_num_ships = rospy.get_param('num_ais_ships', default=5) updated number

if len(self.ships) == updated_num_ships: do nothing
elif len(self.ships) > updated_num_ships: num_remove = len(self.ships) - updated_num_ships. for loop remove ships from self.ships (eg. if had 10 ships but want 2 now, then remove 8 ships from list)
elif len(self.ships) < updated_num_ships: num_add = updated_num_ships - len(self.ships). for loop add ships to self.ships (eg. if had 3 ships but want 7 now, then add 4 random ships to list)